src/components/Checkbox/index.tsx

```diff
@@ -33,8 +33,11 @@ const PaddedText = styled(Text)`
   padding: ${({ theme }) => `0 ${theme.space['0.5']}`};
 `
 
-const StyledIcon = styled.svg<{ size: number }>`
-  margin-right: ${({ theme }) => theme.space['1']};
+const StyledIcon = styled('svg', {
+  shouldForwardProp: prop => !['hasChildren'].includes(prop),
+})<{ size: number; hasChildren: boolean }>`
+  margin-right: ${({ theme, hasChildren }) =>
```

Comment:
Instead of using a margin we can use a gap on parent label as we did for radio component. Using gap will also remove usage of `hasChildren` 😄

Reply:
Done